scripts/ingest-tarball.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# Ingest a tarball containing software, a compatibility layer,
+# or init scripts to the EESSI CVMFS repository, and generate
+# nested catalogs in a separate transaction.
+# This script has to be run on a CVMFS publisher node.
+
+tar_file=$1
+repo=pilot.eessi-hpc.org
+decompress="gunzip -c"
+
+function echo_green() {
+    echo -e "\e[32m$1\e[0m"
+}
+
+function echo_red() {
+    echo -e "\e[31m$1\e[0m"
+}
+
+function error() {
+    echo_red "ERROR: $1" >&2
+    exit 1
+}
+# Check if a tarball has been specified
+if [ "$#" -ne 1 ]; then
+    error "usage: $0 <gzipped tarball>"
+fi
+
+# Check if the given tarball exists
+if [ ! -f "${tar_file}" ]; then
+    error "tar file ${tar_file} does not exist!"
+fi
+
+tar_file_basename=$(basename "${tar_file}")
+version=$(echo ${tar_file} | cut -d- -f2)
+top_level_dir=$(echo ${tar_file} | cut -d- -f3)
+tar_top_level_dir=$(tar tf "${tar_file}" | head -n 1 | cut -d/ -f1)
+
+# Check if the top level dir encoded in the filename
+# matches the top lever dir inside the tarball
+if [ "${top_level_dir}" != "${tar_top_level_dir}" ]
+then
+    error "the top level directory in the filename (${top_level_dir}) does not match the top level directory in the tar ball (${tar_top_level_dir})."
+fi
+
+# We assume that the top level dir must be compat, software, or init
+if [ "${tar_top_level_dir}" != "compat" ] && [ "${tar_top_level_dir}" != "software" ] && [ "${tar_top_level_dir}" != "init" ]
+then
+    error "the top level directory in the tar ball should be either compat, software, or init!"
+fi
+
+# Check of the EESSI version number encoded in the filename
+# is a valid, i.e. matches the format YYYY.DD
+if ! echo "${version}" | egrep -q '^20[0-9][0-9]\.(0[0-9]|1[0-2])$'
+then
+    error "${version} is not a valid EESSI version."
+fi
+
+# Ingest the tarball to the repository
+${decompress} "${tar_file}" | cvmfs_server ingest -t - -b "${version}" "${repo}" && echo_green "${tar_file} ingested to ${repo}"
```

Comment:
Can we include a message on ingest too, like "`ingesting tarball ${tar_file}`"?

Reply:
Added some more output messages and error checks.